### ISSUE TITLE
Upgrade to GraphQL Java 12

### ISSUE THREAD

**Issue by ieugen** [graphql-java-kickstart/graphql-java-tools#249]
Hello,

I've just noticed that GraphQL Java 12 was released. 
Added this issue to track progress for upgrading to graphqk-java 12.

https://github.com/graphql-java/graphql-java/releases/tag/12.0

**Comment by slisaasquatch** [comment]
Just for the record, I tried using 5.5.1 with GraphQL Java 12, and this is the exception I got:
```
java.lang.NoSuchMethodError: graphql.language.Directive.<init>(Ljava/lang/String;Ljava/util/List;Lgraphql/language/SourceLocation;Ljava/util/List;)V
        at com.coxautodev.graphql.tools.relay.RelayConnectionFactory$DirectiveWithField.<init>(RelayConnectionFactory.kt:90)
        at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.withField(RelayConnectionFactory.kt:83)
        at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.directivesWithField(RelayConnectionFactory.kt:87)
        at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.findConnectionDirectives(RelayConnectionFactory.kt:44)
        at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.create(RelayConnectionFactory.kt:25)
        at com.coxautodev.graphql.tools.SchemaParserBuilder.appendDynamicDefinitions(SchemaParserBuilder.kt:178)
        at com.coxautodev.graphql.tools.SchemaParserBuilder.scan(SchemaParserBuilder.kt:167)
        at com.coxautodev.graphql.tools.SchemaParserBuilder.build(SchemaParserBuilder.kt:212)
```

**Comment by rwilliams-r7** [comment]
RootTypeInfo.kt is also needing of a little work. It is using a method that has been renamed. ```OperationTypeDefinition```. See PR https://github.com/graphql-java/graphql-java/pull/1379/files

**Comment by alexgenco** [comment]
Any movement on this? Getting the same exception trying to run graphql-java-tools 5.5.2 against graphql-java 12.0

**Comment by lburja** [comment]
I have created a pull request which fixes this issue:  #262

However, I had to use reflection to access certain fields that are not available anymore. I'm not sure this is acceptable, but I couldn't find an easy fix.

**Comment by ieugen** [comment]
It seems it has been fixed in version 5.6.0 and also with https://github.com/graphql-java-kickstart/graphql-spring-boot/issues/227 with PR https://github.com/graphql-java-kickstart/graphql-spring-boot/pull/237

**Comment by alexgenco** [comment]
We are still running into this with graphql-java-tools 5.6.0 and graphql-java 13.0:

```
graphql.language.Directive.<init>(Ljava/lang/String;Ljava/util/List;Lgraphql/language/SourceLocation;Ljava/util/List;Lgraphql/language/IgnoredChars;)V
	at com.coxautodev.graphql.tools.relay.RelayConnectionFactory$DirectiveWithField.<init>(RelayConnectionFactory.kt:91)
	at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.withField(RelayConnectionFactory.kt:84)
	at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.directivesWithField(RelayConnectionFactory.kt:88)
	at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.findConnectionDirectives(RelayConnectionFactory.kt:45)
	at com.coxautodev.graphql.tools.relay.RelayConnectionFactory.create(RelayConnectionFactory.kt:26)
	at com.coxautodev.graphql.tools.SchemaParserBuilder.appendDynamicDefinitions(SchemaParserBuilder.kt:173)
	at com.coxautodev.graphql.tools.SchemaParserBuilder.scan(SchemaParserBuilder.kt:162)
	at com.coxautodev.graphql.tools.SchemaParserBuilder.build(SchemaParserBuilder.kt:207)
```

**Comment by rrva** [comment]
@alexgenco the error you got occurs with 13.0 not with 12.0. For a possible fix for 13.0, see https://github.com/graphql-java-kickstart/graphql-java-tools/pull/284

**Comment by alexgenco** [comment]
Awesome thanks @rrva we'll give it a shot
